ReactWindows/ReactNative/Modules/Location/LocationModule.cs

```diff
@@ -0,0 +1,194 @@
+                    task, 
+                    Task.Delay(TimeSpan.FromMilliseconds(locationOptions.Timeout)));
+
+                if (completedTask == task)
+                {
+                    var geoposition = await task;
+                    success.Invoke(ConvertGeoposition(geoposition));
+                }
+                else
+                {
+                    error.Invoke("Location request timed out");
+                }
+            }
+            catch (Exception ex)
+            {
+                error.Invoke($"Location request failed with exception: {ex}");
+            }
+        }
+
+        [ReactMethod]
+        public void startObserving(JObject options)
+        {
+            var locationOptions = LocationOptions.FromJson(options);
+            var geolocator = new Geolocator
+            {
+                DesiredAccuracy = locationOptions.HighAccuracy ? PositionAccuracy.High : PositionAccuracy.Default,
+                MovementThreshold = locationOptions.DistanceFilter,
+            };
+
+            var positionObservable = 
+                Observable.FromEventPattern<
+                    TypedEventHandler<Geolocator, PositionChangedEventArgs>,
+                    Geolocator, 
+                    PositionChangedEventArgs>(
+                    h => geolocator.PositionChanged += h,
+                    h => geolocator.PositionChanged -= h);
+
+            var positionQuery = from e in positionObservable
+                                select e.EventArgs.Position;
+
+            var positionSubscription = positionQuery.Subscribe(position =>
+                Context.GetJavaScriptModule<RCTDeviceEventEmitter>()
+                    .emit("geolocationDidChange", ConvertGeoposition(position)));
+
+            var statusObservable =
+                Observable.FromEventPattern<
+                    TypedEventHandler<Geolocator, StatusChangedEventArgs>,
+                    Geolocator,
+                    StatusChangedEventArgs>(
+                        h => geolocator.StatusChanged += h,
+                        h => geolocator.StatusChanged -= h);
+
+            var statusQuery = from e in statusObservable
+                              select e.EventArgs.Status;
+
+            var statusSubscription = statusQuery.Subscribe(status =>
+            {
+                if (status == PositionStatus.Disabled)
+                {
+                    Context.GetJavaScriptModule<RCTDeviceEventEmitter>()
+                        .emit("geolocationError", "Location settings are turned off.");
+                }
+                else if (status == PositionStatus.NotAvailable)
+                {
+                    Context.GetJavaScriptModule<RCTDeviceEventEmitter>()
+                        .emit("geolocationError", "Location services is not available.");
+                }
+                else if (status == PositionStatus.NoData || status == PositionStatus.Initializing)
```

Comment:
do we want a retry if it's just initializing meaning it might take a little bit or is that up to the dev?